File: backend/Lithuaningo.API/Models/Announcement.cs
using System;
using Supabase.Postgrest.Models;
using Supabase.Postgrest.Attributes;

namespace Lithuaningo.API.Models
{
    [Table("announcements")]
    public class Announcement : BaseModel
    {
        [PrimaryKey("id")]
        [Column("id")]
        public Guid Id { get; set; }

        [Column("title")]
        public string Title { get; set; } = string.Empty;

        [Column("content")]
        public string Content { get; set; } = string.Empty;

        [Column("is_active")]
        public bool IsActive { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; }

        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; }

        [Column("valid_until")]
        public DateTime? ValidUntil { get; set; }
    }
}

File: frontend/src/types/Announcement.ts
export interface Announcement {
  id: string;
  title: string;
  content: string;
  isActive: boolean;
  validUntil?: string;
  createdAt: string;
  updatedAt: string;
  status?: string;
  timeRemaining?: string;
}

export interface CreateAnnouncementRequest {
  title: string;
  content: string;
  isActive: boolean;
  validUntil?: string;
}

export interface UpdateAnnouncementRequest {
  title: string;
  content: string;
  isActive: boolean;
  validUntil?: string;
}
